app/services/doc_auth/error_generator.rb

```diff
@@ -103,11 +103,83 @@ def get_image_metric_errors(processed_image_metrics)
     end
   end
 
+  class DocAuthErrorHandler < ErrorHandler
+    def handle(response_info, known_alert_error_count)
+      get_doc_auth_errors(response_info, known_alert_error_count)
+    end
+
+    private
+
+    def get_doc_auth_errors(response_info, known_error_count)
+      # don't worry about unknown alert errors here
+      return if known_error_count < 1
+
+      doc_auth_error_messages = get_doc_auth_error_messages(response_info)
+      liveness_enabled = response_info[:liveness_enabled]
+
+      if known_error_count == 1
+        process_single_doc_auth_error(doc_auth_error_messages)
+      else
+        # Simplify multiple errors into a single error for the user
+        consolidate_multiple_doc_auth_errors(doc_auth_error_messages, liveness_enabled)
+      end
+    end
+
+    def get_doc_auth_error_messages(response_info)
+      errors = Hash.new { |hash, key| hash[key] = Set.new }
+
+      if response_info[:doc_auth_result] != LexisNexis::ResultCodes::PASSED.name
+        response_info[:processed_alerts][:failed]&.each do |alert|
+          alert_msg_hash = ErrorGenerator::ALERT_MESSAGES[alert[:name].to_sym]
+
+          if alert_msg_hash.present?
+            field_type = alert[:side] || alert_msg_hash[:type]
+            errors[field_type.to_sym] << alert_msg_hash[:msg_key]
+          end
+        end
+      end
+      errors
+    end
+
+    def process_single_doc_auth_error(alert_errors)
+      error = alert_errors.values[0].to_a.pop
+      side = alert_errors.keys[0]
+      ErrorResult.new(error, side)
+    end
+
+    def consolidate_multiple_doc_auth_errors(alert_errors, liveness_enabled)
+      error_fields = alert_errors.keys
+      if error_fields.length == 1
+        side = error_fields.first
+        case side
+        when ErrorGenerator::ID
+          error = ErrorGenerator.general_error(liveness_enabled)
+        when ErrorGenerator::FRONT
+          error = Errors::MULTIPLE_FRONT_ID_FAILURES
+        when ErrorGenerator::BACK
+          error = Errors::MULTIPLE_BACK_ID_FAILURES
+        end
+      elsif error_fields.length > 1
+        error = ErrorGenerator.general_error(liveness_enabled)
+        side = ErrorGenerator::ID
+      end
+      ErrorResult.new(error, side)
+    end
+  end
+
   class SelfieErrorHandler < ErrorHandler
     include SelfieConcern
     def handle(response_info)
       liveness_enabled = response_info[:liveness_enabled]
-      get_selfie_error(liveness_enabled, response_info)
+      selfie_error = get_selfie_error(liveness_enabled, response_info)
+
+      if is_generic_selfie_error?(selfie_error)
+        return selfie_general_failure_error
+      else
+        error = selfie_error
+        side = ErrorGenerator::SELFIE
+        return ErrorResult.new(error, side)
```

Comment:
No need for `return` here.

Reply:
Good point! Fixed this in 4496159a5110882609d9f62b671576af9f738dee too.